### PR TITLE
tests: use `parse_metrics` everywhere

### PR DESCRIPTION
**Pull request by problame**
Commits are intended to remain separate.

Review each commit individually. The last one is mostly churn.

### CODE REVIEW

File: test_runner/fixtures/metrics.py [hunk 1]
```diff
@@ -13,7 +13,8 @@ def __init__(self, name: str = ""):
         self.metrics = defaultdict(list)
         self.name = name
 
-    def query_all(self, name: str, filter: Dict[str, str]) -> List[Sample]:
+    def query_all(self, name: str, filter: Optional[Dict[str, str]] = None) -> List[Sample]:
+        filter = filter or {}
```
Comment (hlinnaka) [inline test_runner/fixtures/metrics.py:17]: Could the default value be '{}' directly? I.e.
```
 def query_all(self, name: str, filter: Dict[str, str] = {}) -> List[Sample]:
```

**Review by hlinnaka — APPROVED** [review]
Nice cleanup

File: test_runner/fixtures/metrics.py (hunk 1 above)
Comment (bayandin) [reply test_runner/fixtures/metrics.py:17]: Technically it can, but using mutable data structure as a default value could shoot us in the foot

**Review by bayandin — APPROVED** [review]

File: test_runner/fixtures/metrics.py (hunk 1 above)
Comment (problame) [reply test_runner/fixtures/metrics.py:17]: Yep, I learned this from @vadim2404 a few months back https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

File: test_runner/fixtures/neon_fixtures.py [hunk 2]
```diff
@@ -1434,13 +1434,20 @@ def timeline_download_remote_layers(
                 assert completed["successful_download_count"] > 0
             return completed
 
-    def get_metrics(self) -> str:
+    def get_metrics_str(self) -> str:
+        """You probably want to use get_metrics() instead."""
         res = self.get(f"http://localhost:{self.port}/metrics")
         self.verbose_error(res)
         return res.text
```
[the remark this answers is not in the document]
Comment (problame) [reply test_runner/fixtures/neon_fixtures.py:1441]: The remaining users typically are in places where they also use `Safekeeper.get_metrics_str`. 
Didn't want to cause any inhomogeneity in those places.

File: test_runner/fixtures/neon_fixtures.py [hunk 3]
```diff
@@ -1480,14 +1483,14 @@ def get_remote_timeline_client_metric(
             assert len(matches) < 2, "above filter should uniquely identify metric"
         return value
 
-    def get_metric_value(self, name: str) -> Optional[str]:
+    def get_metric_value(self, name: str) -> Optional[float]:
         metrics = self.get_metrics()
-        relevant = [line for line in metrics.splitlines() if line.startswith(name)]
-        if len(relevant) == 0:
+        results = metrics.query_all(name)
+        if len(results) == 0:
```
Comment (vadim2404) [inline test_runner/fixtures/neon_fixtures.py:1489]: NIT: 

```python
if not results:
```

**Review by vadim2404 — APPROVED** [review]

**Pull request merged**